p2719.md

```diff
@@ -796,53 +852,166 @@ namespace lib {
-the language could always be pursued as a separate proposal since that concern can be tackled orthogonally.
-For example, a special ADL lookup could be done based solely on the dynamic type being [de]allocated.
+There are too many ways in which an unconstrained declaration like this can break, including an
+unexpected set of associated namespaces or even a mere `using namespace lib;`.
+
+Given the need to constrain a type-aware operator anyway, we believe that allowing namespace-scoped
+operators is merely a nice-to-have but not something that we need fundamentally. Furthermore, adding
+this capability to the language could always be pursued as a separate proposal since that concern
+can be tackled orthogonally. For example, a special ADL lookup could be done based solely on the
+dynamic type being [de]allocated.
+
+Since this adds complexity to the proposal and implementation and doesn't provide great value, we
+are not pursuing it as part of this proposal.
+
+## Design choice: Allowing type aware allocation functions in constant expressions
+
+In the current specification it is not possible for an author to specify a constexpr `operator new`
+or `operator delete`, and instead use of `new` and `delete` in constant expressions requires that
+the resolved operator be a replaceable global allocation function [expr.const]{.unnumbered .sref}.
+
+This already causes problems for class scoped operators as the existence of a correctly named
+operator in the class scope prevents any further lookup in the global scope. However it is possible
+for code that wishes to support allocation in constant and custom operators with code of the form:
+
+ ```cpp
+ template <typename T>
+ constexpr int example() {
+    T* value;
+    if consteval {
+        value = ::new T;
+    } else {
+        value = new T;
+    }
+    // Do work
+    if consteval {
+      ::delete value;
+    } else {
+      delete value;
+    }
+    return result;
+}
+ ```
+
+Which, while distasteful and verbose, does work.
+
+However this proposal introduces the possibility, and for some use cases the probability,
+of finding matching operators for non-placement new in the global scope and so being
+unusable in constant expressions. As a result we risk putting developers in the unreasonable
+position of having to choose between two different language features, both of which are
+extremely valuable.
+
+Our solution is to treat type aware global allocators that have no placement parameters as if they
+are replaceable global operators, and as such permit them to be assumed to behave in a like manner.
+In addition to permitting general global type aware allocators without breaking dynamic allocation
+in constant evaluation, this also provides a path to supporting class specialized allocators without
+sacrificing usability in constant evaluation contexts.
+
+Possible alternative approaches, that might also be useful for other contexts (static analysis,
```

Comment:
I would remove this and keep it for a follow-up paper treating constant expressions and `operator new` specifically. There's no need to try solving these adjacent problems in this paper, and that only muddies the waters.